Error during Caffe model conversion with Python3.8 (#6056)
openvinotoolkit/openvino: docs/MO_DG/prepare_model/Model_Optimizer_FAQ.md

  });
</script>

\endhtmlonly

> #### 103. What does the message "ModelOptimizer is not able to parse *.caffemodel" means? <a name="question-103"></a>

Reviewer: means -> mean

> 

Reviewer: also, please double check grammar correctness of sentences.

> If file '*.caffemodel' exist and correct and among possible reasons you can see third point about protobuf implementation then you have used python protobuf implementation. In some cases it gives error during model parsing. Then you should build cpp implementation of protobuf by yourself or use one of wheels from ModelOptimizer package. For complete instructions about building `protobuf` from sources, see the appropriate section in [Converting a Model to Intermediate Representation](Config_Model_Optimizer.md).

Reviewer 2: If file *.caffemodel exist -> If file *.caffemodel exists

> If a file '*.caffemodel' exists and it is correct, you can observe error due to use of python protobuf implementation. In some cases it gives error during model parsing: for example "'utf-8' codec can't decode byte 0xe0 in position 4: invalid continuation byte in field: mo_caffe.SpatialTransformerParameter.transform_type". You can use Python3.6/3.7 or you can build 'cpp' implementation of protobuf by yourself for your version of Python. For complete instructions about building `protobuf` from sources, see the appropriate section in [Converting a Model to Intermediate Representation](Config_Model_Optimizer.md).

Reviewer 3: ```suggestion
If a '*.caffemodel' file exists and it is correct, the error possibly occured due to the use of Python protobuf implementation. In some cases, it shows error message during model parsing, for example: "'utf-8' codec can't decode byte 0xe0 in position 4: invalid continuation byte in field: mo_caffe.SpatialTransformerParameter.transform_type". You can either use Python 3.6/3.7 or build 'cpp' implementation of protobuf yourself for your version of Python. For the complete instructions about building `protobuf` from sources, see the appropriate section in [Converting a Model to Intermediate Representation](Config_Model_Optimizer.md).
```

> If file '*.caffemodel' exists and is correct and among possible reasons you can see the third point about protobuf implementation then you are using python protobuf implementation. In some cases it gives error during model parsing: for example "'utf-8' codec can't decode byte 0xe0 in position 4: invalid continuation byte in field: mo_caffe.SpatialTransformerParameter.transform_type". You can use Python3.6/3.7 or you can build 'cpp' implementation of protobuf by yourself for your version of Python. For complete instructions about building `protobuf` from sources, see the appropriate section in [Converting a Model to Intermediate Representation](Config_Model_Optimizer.md).

Reviewer 4: ```suggestion
You can face Caffe model conversion error (for example densenet-201 Caffe model on Windows* platform with Python* 3.8 environment). It gives an error during model parsing: "'utf-8' codec can't decode byte 0xe0 in position 4: invalid continuation byte in field: mo_caffe.SpatialTransformerParameter.transform_type". In addition, the error message can complain about protobuf implementation when you are using python protobuf implementation. We suggest some workarounds:
1. You can use Python3.6/3.7 to convert a Caffe model on Windows
2. You can build 'cpp' implementation of protobuf by yourself for your version of Python. For complete instructions about building `protobuf` from sources, see the appropriate section in [Converting a Model to Intermediate Representation](Config_Model_Optimizer.md).
```

> #### 104. What does the message "SyntaxError: 'yield' inside list comprehension" during MxNet\* model conversion means? <a name="question-104"></a>

Reviewer: means -> mean

> You can face with "SyntaxError: 'yield' inside list comprehension" issue during converting MxNet\* models (mobilefacedet-v1-mxnet, brain-tumor-segmentation-0001) on Windows* platform with Python* 3.8 environment. This issue existed due to API changes for `yield expression` in Python 3.8 

Reviewer 3: ```suggestion
The issue "SyntaxError: 'yield' inside list comprehension" might occur during converting MXNet\* models (mobilefacedet-v1-mxnet, brain-tumor-segmentation-0001) on Windows* platform with Python* 3.8 environment. This issue is caused by API changes for `yield expression` in Python 3.8.
```

> You can face with "SyntaxError: 'yield' inside list comprehension" issue during converting MxNet\* models (mobilefacedet-v1-mxnet, brain-tumor-segmentation-0001) on Windows* platform with Python* 3.8 environment. This issue existed due to API changes for `yeld expression` in Python 3.8 

Reviewer: yeld -> yield

Reviewer 4: ```suggestion
You can face with "SyntaxError: 'yield' inside list comprehension" issue during converting MxNet\* models (mobilefacedet-v1-mxnet, brain-tumor-segmentation-0001) on Windows* platform with Python* 3.8 environment. This issue existed due to API changes for `yield expression` in Python 3.8 
```

We suggest some workarounds:

> 1. Use Python 3.6/3.7 to convert MxNet\* models on Windows

Reviewer 3: ```suggestion
1. Use Python 3.6/3.7 to convert MXNet\* models on Windows
```

> 2. Update mxnet but in that case, you can have conflicts between previously installed PyPI dependencies: pip install mxnet=1.7.0.post2

Reviewer 3: ```suggestion
2. Update MXNet: pip install mxnet=1.7.0.post2
Note that you might have conflicts between previously installed PyPI dependencies.
```
